### PR TITLE
add post comments to articles by article_id

### PR DESCRIPTION
**Pull request by AlistairHopson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -287,3 +287,59 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+          })
+        );
+      });
+  });
+  test("does not allow comments to be posted by unexisting usernames (i.e. unregistered users)", () => {
+    const testComment = { username: "dave", comment: "I agree!" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(testComment)
+      .expect(401)
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: I would say 404 is more appropriate for usernames that aren't found, 401 indicates forbidden for user to do it but 404 just says not found

Comment (Poonam-raj) [reply __tests__/app.test.js]: But thats my opinion so its not a firm change needed if you believe 401 fits better for you

File: __tests__/app.test.js [hunk 2]
```diff
@@ -287,3 +287,59 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send(testComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toEqual(
+          expect.objectContaining({
+            comment_id: 19,
+            body: "I agree!",
+            article_id: 5,
+            author: "icellusedkars",
+            votes: 0,
```
Comment (Poonam-raj) [inline __tests__/app.test.js:305]: check created_at exists too - you don't have to use the time stamp, simply checking it's a valid string is fine

File: __tests__/app.test.js [hunk 3]
```diff
@@ -287,3 +287,59 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+  });
+  test("users cannot comment on invalid request parameters", () => {
+    const testComment = { username: "icellusedkars", comment: "I agree!" };
+    return request(app)
+      .post("/api/articles/five/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("Invalid data type passed to endpoint.");
+      });
```
Comment (Poonam-raj) [inline __tests__/app.test.js:344]: I'd also want tests for an incorrect post body aka missing required fields.

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
few amendments in testing needed, then throw this over again to us :D

**Comment by AlistairHopson** [comment]
Thanks! I've adjusted these

**Comment by P-Copley** [comment]
Looking good @AlistairHopson 

Go ahead and merge when you're ready. :+1:

**Pull request merged**